core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1355,6 +1396,196 @@ void testDateFormat(DateFormatTester dft) {
+  }
+
+  @ParameterizedTest(name = "{0}{1}")
+  @MethodSource("getTestDataForTimeFormat")
+  public void testTimeFormat(LiteralExpression arg,
+                             LiteralExpression format,
+                             String expectedResult) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    timeFormatQuery(arg, format, expectedResult);
+  }
+
+  private static Stream<Arguments> getInvalidTestDataForTimeFormat() {
+    return Stream.of(
+        Arguments.of(
+            DSL.literal("asdfasdf"),
+            DSL.literal("%f")),
+        Arguments.of(
+            DSL.literal("12345"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("10:11:61"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("10:61:12"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("61:11:12"),
+            DSL.literal("%h"))
+    );
+  }
+  
+  @ParameterizedTest(name = "{0}{1}")
+  @MethodSource("getInvalidTestDataForTimeFormat")
+  public void testInvalidTimeFormat(LiteralExpression arg, LiteralExpression format) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expr = DSL.time_format(functionProperties, arg, format);
+    assertThrows(SemanticCheckException.class, () -> eval(expr));
+  }
+
+  private static Stream<Arguments> getInvalidTimeFormatHandlers() {
+    return Stream.of(
+        Arguments.of("%a"),
+        Arguments.of("%b"),
+        Arguments.of("%j"),
+        Arguments.of("%M"),
+        Arguments.of("%W"),
+        Arguments.of("%D"),
+        Arguments.of("%w"),
+        Arguments.of("%U"),
+        Arguments.of("%u"),
+        Arguments.of("%V"),
+        Arguments.of("%v"),
+        Arguments.of("%X"),
+        Arguments.of("%x")
+    );
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("getInvalidTimeFormatHandlers")
+  public void testTimeFormatWithInvalidHandlers(String handler) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expr = DSL.time_format(
+        functionProperties,
+        DSL.literal("12:23:34"),
+        DSL.literal(handler));
+    assertEquals(ExprNullValue.of(), eval(expr));
+  }
+
+  @Test
+  public void testTimeFormatWithDateHandlers() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expr = DSL.time_format(
+        functionProperties,
+        DSL.literal(new ExprDateValue("2023-01-17")),
+        DSL.literal("%c %d %e %m %Y %y"));
+    assertEquals(
+        "0 00 0 00 0000 00",
+        eval(expr).stringValue());
+  }
+
```

Comment:
Add test to compare `date_format` to `time_format` when using format specifiers for hours/minutes/seconds.